testing/docker/elastic-agent/build.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/bash
+
+# This script builds an image from the elastic-agent image
+# with a locally built apm-server binary injected. Additional
+# flags (e.g. -t <name>) will be passed to `docker build`.
+
+set -eu
+
+REPO_ROOT=$(cd $(dirname $(realpath "$0"))/../../.. && pwd)
+
+DEFAULT_IMAGE_TAG=$(grep docker.elastic.co/kibana ${REPO_ROOT}/docker-compose.yml | cut -d: -f3)
+BASE_IMAGE="${BASE_IMAGE:-docker.elastic.co/beats/elastic-agent:$DEFAULT_IMAGE_TAG}"
+GOARCH="${GOARCH:-$(go env GOARCH)}"
```

Comment:
This will result in in arm64 images being created from arm machines. The deployment template that we're using in Terraform is `gcp-compute-optimized-v2`, which uses an `amd64` architecture. Shouldn't we always default to building amd64 images and binaries for this particular workflow?

Reply:
Thanks, good catch. I'll just pass this in from the system tests.

Reply:
Ah, actually: the Makefile specifies GOARCH=amd64 for `build_elastic_agent_docker_image` target.